tests-functional/tests/test_contact_request.py

```diff
@@ -0,0 +1,99 @@
+from time import sleep
+from uuid import uuid4
+import pytest
+from test_cases import OneToOneMessageTestCase
+from constants import DEFAULT_DISPLAY_NAME
+from clients.signals import SignalType
+
+@pytest.mark.rpc
+class TestContactRequests(OneToOneMessageTestCase):
+
+
+    @pytest.mark.dependency(name="test_contact_request_baseline")
+    def test_contact_request_baseline(self, contact_request_count=1):
+
+        await_signals = [
+            SignalType.MESSAGES_NEW.value,
+            SignalType.MESSAGE_DELIVERED.value,
+        ]
+
+        for i in range(contact_request_count):
+
+            message_text = f"test_contact_request_{i+1}_{uuid4()}"
+
+            sender = self.initialize_backend(await_signals=await_signals)
+            receiver  = self.initialize_backend(await_signals=await_signals)
+
+            pk_sender = sender.get_pubkey(DEFAULT_DISPLAY_NAME)
+            pk_receiver = receiver.get_pubkey(DEFAULT_DISPLAY_NAME)
+
+            existing_contacts = receiver.get_accounts()
+
+            response = sender.send_contact_request([{"id": pk_receiver, "message": message_text}])
+
+            messages_new_event = receiver.find_signal_containing_pattern(SignalType.MESSAGES_NEW.value, event_pattern=message_text, timeout=60)
```

Comment:
It's better to wait for `message_id` rather then text. 
You can get the message id from send contact request `response`.

Reply:
`result/messages[]/id` is the right one
There must be exactly one message, and this must be the message you sent

Reply:
Fair point, indeed in this case you'll have 2 messages.
Yet, we should still refer to `result/messages`, rather then `result/chats/lastMessage`.

# Solution

The solution is to also check the `message.contentType` to be `11`: https://github.com/status-im/status-go/blob/4aa683f03ddf8e71a909328786533a081e584b79/protocol/protobuf/chat_message.proto#L260

The second message "You sent a contact request to..." has `contentType: 15`, which is: https://github.com/status-im/status-go/blob/4aa683f03ddf8e71a909328786533a081e584b79/protocol/protobuf/chat_message.proto#L265-L266

This is a "system message", it only exists locally and is never sent to network.

You can find all content types here:
https://github.com/status-im/status-go/blob/4aa683f03ddf8e71a909328786533a081e584b79/protocol/protobuf/chat_message.proto#L246-L273

# Why am I pushing for usage of `result.messages`?

Because of what we want to test.

| Property | Existence reason | 

WHEN Alice sends a contact request to Bob
THEN A contact request message is sent to Bob
AND a system message "contact request sent to ..." is show to Alice
AND last message in chat is updated
AND activity center notification is generated

Each of `result.messages` / `.chats[].lastMessage` / `.activityCenterNotifications` exists to solve one of the above requirements. So each requirement must be checked by according property in the signal.

Even we remove requirements for system messages, last message and ac notifications, tests should continue to work properly checking the `result.messages`. Hope this clarifies things 🙂